pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/AbstractApexNodeBase.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.ast;
+
+import net.sourceforge.pmd.lang.ast.AbstractNode;
+import net.sourceforge.pmd.lang.ast.Node;
+import net.sourceforge.pmd.lang.ast.SourceCodePositioner;
+
+public abstract class AbstractApexNodeBase extends AbstractNode {
+
+    public AbstractApexNodeBase(int id) {
+        super(id);
+    }
+
+    public AbstractApexNodeBase(Class<?> klass) {
+        super(klass.hashCode());
+    }
+
+    public void calculateLineNumbers(SourceCodePositioner positioner, int startOffset, int endOffset) {
```

Comment:
This method should not be public - it's enough, if it can be called from classes within the same package (`net.sourceforge.pmd.lang.apex.ast`).